### PR TITLE
feat: return all (semver) tags sharing one commit

### PR DESCRIPTION
**Pull request by dinoboff**
When one commit has more than one semver tag, do not pick one but emit both with same the hash.

### CODE REVIEW

**Comment by dinoboff** [comment]
@ikhemissi It's a bigger PR than it should be but` getList` would have getting complicated to follow with that array of array to flatten in the middle.

I refactored it in the first commit and then changed the parsing of the list of tags.

I hope it's ok.

File: src/index.js
Comment (ikhemissi) [inline src/index.js]: I would say Query or extract semantic version tags from a local git repository, instead of github

File: src/index.js
Comment (ikhemissi) [inline src/index.js]: Would you please refactor the function to return the result (tags) at the end, for readability purpose ?
It is not a long function, so my opinion may not be relevant, but if someone else updates the function it may become complicated very fast.

For example, something like 
```javascript
function getSemanticTags(refs) {
  const tags = [];

  while (true) {
    const match = tagRegex.exec(refs);

    if (match == null) {
      break;
    }

    const tag = match[1];
    const version = semver.valid(tag);

    if (version != null) {
      tags.push({ tag, version });
    }
  }

  return tags;
}
```

or (without the linting warning, but with line duplication):
```javascript
function getSemanticTags(refs) {
  const tags = [];
  let match = tagRegex.exec(refs);

  while (match) {
    const tag = match[1];
    const version = semver.valid(tag);

    if (version != null) {
      tags.push({ tag, version });
    }

    match = tagRegex.exec(refs);
  }

  return tags;
}
```

File: src/index.js
Comment (ikhemissi) [inline src/index.js]: Just nitpicking: `getSemanticTags` is returning the list of tags and versions from a particular commit, so how about renaming it to something like `extractCommitTagsAndVersions` ?

`const tagsAndVersions = extractCommitTagsAndVersions(match[1]);` ?!

**Review by ikhemissi — CHANGES_REQUESTED** [review]
Thank you very much @dinoboff  for the excellent contribution (as always).

I made a couple of comments (which may be a bit opinionated), well more like questions.

By the way, what was the use case for which you needed to handle multiple semantic versions for a single commit ?
Was it because of pre-releases (http://semver.org/#spec-item-9), or something similar ?

**Comment by dinoboff** [comment]
Although, most repo will have a file holding the version number (be it "package.json" for a npm package or "VERSION" for something more generic) that's not a requirement. You could have a commit having more than one semantic tag, e.g. a bug free RC version becoming final version.

**Comment by dinoboff** [comment]
@ikhemissi Thanks for the review. It should be fixed.

Please, let me rebase before merging once there's nothing else to fix.

File: src/index.js
Comment (dinoboff) [reply src/index.js]: It was a lapsus, Thanks

**Review by ikhemissi — APPROVED** [review]

**Comment by ikhemissi** [comment]
I'm happy with the changes and I will merge them (and create a release) once you finish the WIP.
Thank you @dinoboff

**Comment by ikhemissi** [comment]
Released in version 1.3.0

**Pull request merged**
